homeassistant/components/netatmo/sensor.py

```diff
@@ -541,83 +510,24 @@ def update(self):
 class NetatmoData:
     """Get the latest data from Netatmo."""
 
-    def __init__(self, auth, data_class, station):
+    def __init__(self, auth, station_data):
         """Initialize the data object."""
-        self.auth = auth
-        self.data_class = data_class
         self.data = {}
-        self.station_data = self.data_class(self.auth)
-        self.station = station
-        self.station_id = None
-        if station:
-            station_data = self.station_data.stationByName(self.station)
-            if station_data:
-                self.station_id = station_data.get("_id")
+        self.station_data = station_data
         self._next_update = time()
-        self._update_in_progress = threading.Lock()
+        self.auth = auth
 
     def get_module_infos(self):
         """Return all modules available on the API as a dict."""
-        if self.station_id is not None:
-            return self.station_data.getModules(station_id=self.station_id)
         return self.station_data.getModules()
 
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
-        """Call the Netatmo API to update the data.
+        """Call the Netatmo API to update the data."""
+        self.station_data = self.station_data.__class__(self.auth)
```

Comment:
Why do we need to create a new station?

Reply:
Because that's how pyatmo is designed right now. I am discussing changes with the maintainer but that wont happen in the next days. But I am totally with you on that. It is not pretty.